velox/expression/CoalesceExpr.cpp

```diff
@@ -107,14 +108,48 @@ TypePtr CoalesceCallToSpecialForm::resolveType(
   return CoalesceExpr::resolveType(argTypes);
 }
 
+ExprPtr CoalesceCallToSpecialForm::optimize(
```

Comment:
The logic here is not easy to read. Would you add a comment to explain what this function does to help future readers?